### PR TITLE
Richards mechanics saturation dependent swelling

### PR DESCRIPTION
**Pull request by endJunction**
MPL: implementation of swelling stress rate.
RM: add the swelling stress if the swelling stress is given in the project file.

See individial commits for details.

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.0)
2. [x] Tests covering your feature were added?
3. [x] Any new feature or behavior change was documented?

### CODE REVIEW

**Comment by endJunction** [comment]
@wenqing In the last rebase I added the derivative d\sigma / dS_L to the property and the corresponding part in the jacobian. You might want to have a look on these too.

**Comment by wenqing** [comment]
@endJunction  The changes about  `d\sigma / dS_L` is good.

File: MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include "MaterialLib/MPL/Properties/SaturationDependentSwelling.h"
+
+#include <algorithm>
+#include <iostream>
```
Comment (TomFischer) [inline MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp]: Is this include necessary?

File: MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+#include "MaterialLib/MPL/Medium.h"
+#include "MathLib/KelvinVector.h"
+#include "ParameterLib/CoordinateSystem.h"
+
+namespace MaterialPropertyLib
+{
+SaturationDependentSwelling::SaturationDependentSwelling(
+    std::array<double, 3> swelling_pressures,
+    std::array<double, 3>
```
Comment (TomFischer) [inline MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp]: Is this clang formatted?

File: MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp (hunk 1 above)
Comment (endJunction) [reply MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp]: oh, no. My favorite ;)
Dropped also the algorithm...

File: MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp (hunk 2 above)
Comment (endJunction) [reply MaterialLib/MPL/Properties/SaturationDependentSwelling.cpp]: clang-9.0.1 does it this way, unfortunately.

**Pull request merged**